kernel/arch/dreamcast/fs/fs_dclsocket.c

```diff
@@ -206,22 +206,14 @@ static void dcls_recv_loop(void) {
 }
 
 static void *dcls_open(struct vfs_handler *vfs, const char *fn, int mode) {
-    int hnd, locked;
-    int dcload_mode = 0;
+    int hnd, dcload_mode = 0;
     int mm = (mode & O_MODE_MASK);
     command_t *cmd = (command_t *)pktbuf;
 
     (void)vfs;
 
-    locked = mutex_trylock(&mutex);
-
-    if(locked == -1 && irq_inside_int()) {
-        errno = EAGAIN;
-        return 0;
-    }
-    else if(locked == -1) {
-        mutex_lock(&mutex);
-    }
+    if(mutex_lock_irqsafe(&mutex))
+        return NULL;
```

Comment:
Should we be setting `errno = EAGAIN` here?

Reply:
There are other cases of this below, so assume I've written this comment on those too :p